### PR TITLE
docs(tutorial): part 4

### PR DESCRIPTION
**Pull request by wardbell**
Should be ready to go. Needs a sanity check.

### CODE REVIEW

**Comment by wardbell** [comment]
Where did googlebot get the crazy idea that anyone other than John and Me had something to do with this?

**Comment by PascalPrecht** [comment]
> We found a Contributor License Agreement for you (the sender of this pull request) and all commit authors, but as best as we can tell these commits were authored by someone else

All it says is that you're sending a PR with commits that aren't crafted by you. :)

File: public/docs/_examples/toh-4/ts/app/app.component.1.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  </div>
+  <my-hero-detail [hero]="selectedHero"></my-hero-detail>
+  `,
+  directives: [HeroDetailComponent],
+  // #docregion providers
+  providers: [HeroService]
+  // #enddocregion providers
+})
+// #docregion on-init
+export class AppComponent implements OnInit {
```
Comment (PascalPrecht) [inline public/docs/_examples/toh-4/ts/app/app.component.1.ts]: Isn't this `ngOnInit` now?

Comment (PascalPrecht) [inline public/docs/_examples/toh-4/ts/app/app.component.1.ts]: Ah no. It was that thing where the interface is called differently than the implementation....

**Comment by PascalPrecht** [comment]
Looks like this commit introduces changes in other parts of the tutorial as well

**Comment by PascalPrecht** [comment]
LGTM otherwise

**Comment by wardbell** [comment]
Yes, it touches pt3 (which can now refer to pt4) and updates the final tutorial code with the simplifications I found in pt4

File: public/docs/_examples/toh-4/ts/app/app.component.1.ts (hunk 1 above)
Comment (johnpapa) [inline public/docs/_examples/toh-4/ts/app/app.component.1.ts]: yes. that always confused me too

**Pull request closed without merging**